### PR TITLE
Migrating utils/user_agent.jsx to TypeScript

### PR DESCRIPTION
**Pull request by NathanPickard**
#### Summary

Migrating utils/user_agent.jsx to TypeScript

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/12482

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -5455,6 +5455,7 @@
           "resolved": "https://registry.npmjs.org/p-timeout/-/p-timeout-2.0.1.tgz",
```
Comment (jespino) [inline package-lock.json]: This changes are not related to this PR, do you mind to remove them?

File: actions/command.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ import * as PostActions from 'actions/post_actions.jsx';
 
 import {isUrlSafe, getSiteURL} from 'utils/url.jsx';
 import {localizeMessage, getUserIdFromChannelName} from 'utils/utils.jsx';
-import * as UserAgent from 'utils/user_agent.jsx';
+import * as UserAgent from 'utils/user_agent.tsx';
```
Comment (jespino) [inline actions/command.js]: We prefer to use imports without the extension for typescript, do you mind to change it everywhere by `utils/user_agent`?

Comment (NathanPickard) [reply actions/command.js]: Will make changes, thanks!
